### PR TITLE
Add Horizontal list of dispute details to the transaction page

### LINKED ISSUE #6924: Transaction Details > Dispute Details > Dispute Details
### Description

Add `Dispute Details` section to the `DisputeDetails` component on the transaction screen.

This is part of a larger project to include the Dispute Details on the transaction screen rather than having a separate page.

### Screenshot

Screenshot 2023-08-04 at 09 20 12 copy

### Detailed Changes

A row layout will be added to the `DisputeDetails` Component showing the key attributes of the dispute. This will show at all times regardless of the status of the dispute.

- **Dispute Amount** in the store currency. If the store accepts multicurrency payments and the customer paid `$70.29 AUD`. The amount would be shown as `$45.45 USD`
- **Disputed On** The date the dispute was opened
- **Reason** The reason for the dispute.
  - This includes a `ClickTooltip` component showing more information about the reason on click.
- **Respond By** The evidence required by date.
  - Default: Shows date/time with number of days left `Aug 18, 2023 11:59 PM (12 days left to respond)`
  - Less than 7 days: As above but text showing days left is yellow `(6 days left to respond)` `WP Yellow/Yellow 30 ( #DBA617 )` 
  - Less than 3 days: As above but text showing days left is red `(2 days left to respond)` `WP Gutenberg/Alert Red ( ##CC1818 )` 
  - Less than 1 day: Same as `Less than 3 days` but text now reads `(Last day today)`

No details are shown once evidence is submitted or the response date has passed.

### Additional Screenshots

**Less than 7 days remaining**
image

**Less than 3 days remaining**
image

**Last Day**
image

**Currency must match store currency**
image

### LINKED ISSUE #7164: DisputeDetails JS test failing due to elapsed `due_by` date
### Describe the bug

The JS test for `DisputeDetails → correctly renders dispute details` fails due to a `due_by` date that has elapsed. See error below:

```
TestingLibraryElementError: Unable to find an element with the text: /The cardholder claims this is an unauthorized transaction/. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.
```

This test should be corrected by being provided a dispute with a `due_by` date that is in the future at the time of testing.

https://github.com/Automattic/woocommerce-payments/blob/69a25f7a57c13ca517c29d600d92c393791cd011/client/payment-details/dispute-details/test/index.test.tsx#L89

### To Reproduce

1. Run JS tests for `npm run test:js "client/payment-details/dispute-details"`.
2. Observe the failing test.

### Additional details
This test was introduced in #7139.

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #6924
Fixes #7164

#### Changes proposed in this Pull Request

Screenshot 2023-09-01 at 3 26 28 PM

A row layout will be added to the DisputeDetails Component showing the key attributes of the dispute. This will show at all times regardless of the status of the dispute.

 - `Dispute Amount` ~in the store currency. If the store accepts multicurrency payments and the customer paid $70.29 AUD. The amount would be shown as $45.45 USD~
 - `Disputed On` The date the dispute was opened Reason The reason for the dispute. This includes a ClickTooltip component showing more information about the reason on click.
 - `Respond By` The evidence required by date.
   - Default: Shows date/time with number of days left Aug 18, 2023 11:59 PM (12 days left to respond)
   - Less than 7 days: As above but text showing days left is yellow (6 days left to respond) WP Yellow/Yellow 30 ( #DBA617 )
   - Less than 3 days: As above but text showing days left is red (2 days left to respond) WP Gutenberg/Alert Red ( ##CC1818 )
   - Less than 1 day: Same as Less than 3 days but text now reads (Last day today)
   - 
No details are shown once evidence is submitted or the response date has passed.
 
Screenshot 2023-09-01 at 3 26 38 PM

#### Testing instructions

1. Enable feature flag by running `update_option( '_wcpay_feature_dispute_on_transaction_page', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/) or by modifying your database table `wp_options` directly.
2. Load a Transaction Details page of an active dispute (`/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions%2Fdetails&id=pi_abc&transaction_id=txn_abc&transaction_type=charge`).
3. See the Dispute Details. 
4. Test with differing currencies, dispute reasons and respond by dates.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7077` or branch name `add/6924-dispute-details-attributes` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/6924-dispute-details-attributes) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: d01a689198342bb04cf1677febc8bf2dd1f7b273
- Build time: 2023-09-11 03:04:07 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-details/dispute-details/dispute-summary-row.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+	const disputeReason = formatStringValue(
+		reasons[ dispute.reason ]?.display || dispute.reason
+	);
+	const disputeReasonSummary = reasons[ dispute.reason ]?.summary || [];
+
+	const columns = [
+		{
+			title: __( 'Dispute Amount', 'woocommerce-payments' ),
+			content: formatCurrency( dispute.amount, dispute.currency ),
```
Comment (brucealdridge) [inline client/payment-details/dispute-details/dispute-summary-row.tsx:45]: The issue and mockups specify that this should be in the store's currency. Currently, I have this is set to the dispute currency. 

IMO we should keep these amounts in the same currency and I believe changing this headline amount is out of scope.

image

File: client/payment-details/dispute-details/dispute-summary-row.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		? dateI18n(
+				'M j, Y, g:ia',
+				moment( dispute.evidence_details?.due_by * 1000 ).toISOString()
+		  )
+		: '–';
+
+	const disputeReason = formatStringValue(
+		reasons[ dispute.reason ]?.display || dispute.reason
+	);
+	const disputeReasonSummary = reasons[ dispute.reason ]?.summary || [];
```
Comment (brucealdridge) [inline client/payment-details/dispute-details/dispute-summary-row.tsx:40]: Using the summary seems like the best option from the strings file without adding new content.

File: client/payment-details/dispute-details/dispute-summary-row.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+				: '–',
+		},
+		{
+			title: __( 'Reason', 'woocommerce-payments' ),
+			content: (
+				<>
+					{ disputeReason }
+					{ disputeReasonSummary.length > 0 && (
+						<ClickTooltip
+							buttonIcon={ <HelpOutlineIcon /> }
```
Comment (brucealdridge) [inline client/payment-details/dispute-details/dispute-summary-row.tsx:63]: This icon doesn't align correctly. The way it is rendered inside `ClickTooltip` looks like changing is going to be complicated.

Mockup vs implemented.
Screenshot 2023-09-01 at 4 12 13 PM
Screenshot 2023-09-01 at 4 12 18 PM

**Comment by shendy-a8c** [comment]
Closed and re-opened PR to re-trigger the GH checks.
Update: [GH is currently having a problem with Actions](https://www.githubstatus.com/incidents/tqrmcdmghldw).

File: client/payment-details/dispute-details/style.scss [hunk 4]
```diff
@@ -7,5 +7,49 @@
...
+				align-items: flex-start;
+			}
+			.woocommerce-list__item-title {
+				text-transform: uppercase;
+				color: $gray-700;
+				font-size: 11px;
+				font-style: normal;
+				font-weight: 600;
+				line-height: 16px; /* 145.455% */
+			}
```
Comment (Jinksi) [inline client/payment-details/dispute-details/style.scss]: This might solve the ClickTooltip vertical alignment issue.
```diff
			.woocommerce-list__item-title {
				text-transform: uppercase;
				color: $gray-700;
				font-size: 11px;
				font-style: normal;
				font-weight: 600;
				line-height: 16px; /* 145.455% */
			}
+			.woocommerce-list__item-content {
+				display: flex;
+			}
```

image

File: client/payment-details/dispute-details/dispute-summary-row.tsx (hunk 3 above)
Comment (Jinksi) [reply client/payment-details/dispute-details/dispute-summary-row.tsx:63]: I've added a suggestion that may solve this https://github.com/Automattic/woocommerce-payments/pull/7077#discussion_r1315233606

File: client/payment-details/dispute-details/style.scss [hunk 5]
```diff
@@ -7,5 +7,49 @@
 
 	.transaction-details-dispute-details-body {
 		padding: $grid-unit-20;
+
+		.dispute-summary-row {
+			margin: 24px 0;
+			.woocommerce-list__item-text {
```
Comment (Jinksi) [inline client/payment-details/dispute-details/style.scss]: While working on #6974, I realise that I'll be making these same changes.

Perhaps these changes should be applied WooPayments-wide to the [`HorizontalList` component CSS](https://github.com/Automattic/woocommerce-payments/blob/9dcc66d2c29e73c1674465d7caf8fc4774009d2b/client/components/horizontal-list/style.scss)?

Comment (Jinksi) [reply client/payment-details/dispute-details/style.scss]: **Update**: the `HorizontalList` list component is **only** used on this screen, so we should change the styles within the component's CSS file.

https://github.com/Automattic/woocommerce-payments/blob/9dcc66d2c29e73c1674465d7caf8fc4774009d2b/client/payment-details/summary/index.tsx#L32

File: client/payment-details/dispute-details/dispute-summary-row.tsx (hunk 1 above)
Comment (Jinksi) [reply client/payment-details/dispute-details/dispute-summary-row.tsx:45]: > IMO we should keep these amounts in the same currency and I believe changing this headline amount is out of scope.

I was looking at updating this as part of #6974 – I agree though, I'll open a separate issue for the headline amount.

Comment (Jinksi) [reply client/payment-details/dispute-details/dispute-summary-row.tsx:45]: Issue #7124 opened and slack discussion about scope here: p1693885409513359-slack-C03LA109BKM

File: client/payment-details/dispute-details/style.scss (hunk 5 above)
Comment (Jinksi) [reply client/payment-details/dispute-details/style.scss]: Since this is required across 2 PRs and will be a simple change, I'll create a new PR for the style changes to `HorizontalList`. See #7126.

File: client/payment-details/dispute-details/style.scss (hunk 4 above)
Comment (Jinksi) [reply client/payment-details/dispute-details/style.scss]: Also added this suggestion to #7126

File: client/payment-details/dispute-details/dispute-summary-row.tsx (hunk 3 above)
Comment (Jinksi) [reply client/payment-details/dispute-details/dispute-summary-row.tsx:63]: #7126 has been merged, which should resolve this alignment issue.

File: client/payment-details/dispute-details/style.scss (hunk 5 above)
Comment (Jinksi) [reply client/payment-details/dispute-details/style.scss]: #7126 has been merged, which should allow most of this CSS to be removed 💣

File: client/payment-details/dispute-details/style.scss (hunk 4 above)
Comment (Jinksi) [reply client/payment-details/dispute-details/style.scss]: #7126 has been merged, which should resolve this alignment issue.

File: client/payment-details/dispute-details/style.scss [hunk 6]
```diff
@@ -7,5 +7,35 @@
...
+				gap: var( --grid-unit-05, 4px );
+				.dispute-warning {
+					color: $wp-yellow-30;
+					font-weight: 700;
+				}
+				.dispute-urgent {
+					font-weight: 700;
+					color: $alert-red;
+				}
+			}
```
Comment (Jinksi) [inline client/payment-details/dispute-details/style.scss]: I think we should opt for BEM-style classNames here, to avoid style conflicts and be more consistent with WCPay code. E.g. use `.dispute-summary-row__response-date--warning` rather than `.dispute-summary-row .response-date.dispute-warning`.

```suggestion
			&__response-date {
				display: flex;
				align-items: center;
				gap: var( --grid-unit-05, 4px );
				&--warning {
					color: $wp-yellow-30;
					font-weight: 700;
				}
				&--urgent {
					font-weight: 700;
					color: $alert-red;
				}
			}
```

Also, (nit) `.dispute-summary-row` could be moved to the top-level rather than nested 2 levels deep under `.transaction-details-dispute-details-wrapper` to improve readability and composability.

File: client/payment-details/dispute-details/dispute-summary-row.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+							buttonLabel={ __(
+								'Learn more',
+								'woocommerce-payments'
+							) }
+							content={
+								<div className="dispute-reason-tooltip">
+									<p>{ disputeReason }</p>
+									<Paragraphs>
```
Comment (Jinksi) [inline client/payment-details/dispute-details/dispute-summary-row.tsx:71]: TIL: `<Paragraphs />` component exists!

File: client/payment-details/dispute-details/style.scss [hunk 8]
```diff
@@ -7,5 +7,35 @@
...
 	.transaction-details-dispute-details-body {
 		padding: $grid-unit-20;
+
+		.dispute-summary-row {
+			margin: 24px 0;
+
+			.response-date {
+				display: flex;
+				align-items: center;
+				gap: var( --grid-unit-05, 4px );
```
Comment (Jinksi) [inline client/payment-details/dispute-details/style.scss:25]: I suggest allowing the response-date elements to wrap, to aid readability on <1200px wide viewports.
```suggestion
				gap: var( --grid-unit-05, 4px );
				flex-wrap: wrap;
```

**Before**
image


**After**
image

File: client/payment-details/dispute-details/dispute-summary-row.tsx [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+										>
+											{ __(
+												'Learn more',
+												'woocommerce-payments'
+											) }
+										</a>
+									</p>
+								</div>
+							}
+						></ClickTooltip>
```
Comment (Jinksi) [inline client/payment-details/dispute-details/dispute-summary-row.tsx]: Nit: no children are provided, so we can use a self-closing tag.
```suggestion
						/>
```

File: client/payment-details/dispute-details/dispute-summary-row.tsx [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+			content: (
+				<span className="response-date">
+					{ respondByDate }
+					<span
+						className={
+							( daysRemaining < 3 ? 'dispute-urgent ' : '' ) +
+							( daysRemaining < 7 && daysRemaining > 2
+								? 'dispute-warning '
+								: '' )
+						}
```
Comment (Jinksi) [inline client/payment-details/dispute-details/dispute-summary-row.tsx]: Suggestion: use the `classnames` library to improve readability here.
```suggestion
						className={ classNames( {
							'dispute-urgent': daysRemaining < 3,
							'dispute-warning':
								daysRemaining < 7 && daysRemaining > 2,
						} ) }
```

**Review by Jinksi — APPROVED** [review]
This PR tests well @brucealdridge 🏆 

I've left some recommended code improvements but these do not block merging, IMO.

**I've tested the following**:

* Various dispute reason tooltips, including General, Duplicate and some without descriptions/summaries (`check_returned`). ✅
* Within 24h ✅
	image
* Within 72h ✅
	image
* Within 7 days ✅
	image
* \> 7 days ✅
	image
* Renders well on mobile (iOS Safari 16.6) ✅

**Comment by Jinksi** [comment]
@brucealdridge I've resolved the merge conflicts introduced when #7139 was merged to `develop` in bed4127.

File: client/payment-details/dispute-details/dispute-summary-row.tsx (hunk 10 above)
Comment (brucealdridge) [reply client/payment-details/dispute-details/dispute-summary-row.tsx]: TIL! thanks

**Comment by Jinksi** [comment]
@brucealdridge I have written some component tests for this feature we could merge into this branch if you like the look: #7144. Or it could be reviewed separately and merged to `develop` afterwards.
